packages/mui-base/src/AlertDialog/Backdrop/AlertDialogBackdrop.tsx

```diff
@@ -38,17 +38,12 @@ const AlertDialogBackdrop = React.forwardRef(function AlertDialogBackdrop(
   forwardedRef: React.ForwardedRef<HTMLDivElement>,
 ) {
   const { render, className, keepMounted = false, ...other } = props;
-  const { open, hasParentDialog, setBackdropPresent, animated } = useAlertDialogRootContext();
-
-  const handleMount = React.useCallback(() => setBackdropPresent(true), [setBackdropPresent]);
```

Comment:
I didn't bother refactoring this as we won't need this logic anyway after https://github.com/mui/base-ui/issues/623